plugins/modules/mso_tenant.py

```diff
@@ -108,59 +125,63 @@
 def main():
     argument_spec = mso_argument_spec()
     argument_spec.update(
-        description=dict(type='str'),
-        display_name=dict(type='str'),
-        tenant=dict(type='str', aliases=['name']),
-        users=dict(type='list', elements='str'),
-        sites=dict(type='list', elements='str'),
-        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+        description=dict(type="str"),
+        display_name=dict(type="str"),
+        tenant=dict(type="str", aliases=["name"]),
+        users=dict(type="list", elements="str"),
+        sites=dict(type="list", elements="str"),
+        orchestrator_only=dict(type="str", default="yes", choices=["yes", "no"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
     )
 
     module = AnsibleModule(
         argument_spec=argument_spec,
         supports_check_mode=True,
         required_if=[
-            ['state', 'absent', ['tenant']],
-            ['state', 'present', ['tenant']],
+            ["state", "absent", ["tenant"]],
+            ["state", "present", ["tenant"]],
         ],
     )
 
-    description = module.params.get('description')
-    display_name = module.params.get('display_name')
-    tenant = module.params.get('tenant')
-    state = module.params.get('state')
+    description = module.params.get("description")
+    display_name = module.params.get("display_name")
+    tenant = module.params.get("tenant")
+    orchestrator_only = module.params.get("orchestrator_only")
+    state = module.params.get("state")
 
     mso = MSOModule(module)
 
     # Convert sites and users
-    sites = mso.lookup_sites(module.params.get('sites'))
-    users = mso.lookup_users(module.params.get('users'))
+    sites = mso.lookup_sites(module.params.get("sites"))
+    users = mso.lookup_users(module.params.get("users"))
 
     tenant_id = None
-    path = 'tenants'
+    path = "tenants"
 
     # Query for existing object(s)
     if tenant:
         mso.existing = mso.get_obj(path, name=tenant)
         if mso.existing:
-            tenant_id = mso.existing.get('id')
+            tenant_id = mso.existing.get("id")
             # If we found an existing object, continue with it
-            path = 'tenants/{id}'.format(id=tenant_id)
+            path = "tenants/{id}".format(id=tenant_id)
     else:
         mso.existing = mso.query_objs(path)
 
-    if state == 'query':
+    if state == "query":
         pass
 
-    elif state == 'absent':
+    elif state == "absent":
         mso.previous = mso.existing
         if mso.existing:
             if module.check_mode:
                 mso.existing = {}
             else:
-                mso.existing = mso.request(path, method='DELETE')
+                orchestrator_only_map = {"yes": "true", "no": "false"}
```

Comment:
Should we define this as a generic constant for toggling true/false to yes/no and other way around? I did create a constants file where we could place these. 

https://github.com/CiscoDevNet/ansible-mso/blob/master/plugins/module_utils/constants.py